src/analysis/convex/between.lean

```diff
@@ -355,6 +355,37 @@ begin
   rw (line_map_injective R hxy).mem_set_image
 end
 
+omit V
+
+@[simp] lemma wbtw_mul_sub_add_iff [no_zero_divisors R] {x y r : R} :
+  wbtw R x (r * (y - x) + x) y ↔ x = y ∨ r ∈ set.Icc (0 : R) 1 :=
+wbtw_line_map_iff
+
+@[simp] lemma sbtw_mul_sub_add_iff [no_zero_divisors R] {x y r : R} :
+  sbtw R x (r * (y - x) + x) y ↔ x ≠ y ∧ r ∈ set.Ioo (0 : R) 1 :=
+sbtw_line_map_iff
+
+@[simp] lemma wbtw_zero_one_iff {x : R} : wbtw R 0 x 1 ↔ x ∈ set.Icc (0 : R) 1 :=
```

Comment:
Is something like `wbtw R a x b ↔ x ∈ set.Icc (min a b) (max a b)` true?

Reply:
I think that would work in a `linear_ordered_field`; not in an `ordered_ring` which is what applies here. (No idea how useful anything in convexity is without a field (it's probably a case where there are multiple sensible definitions that are all equivalent for fields and not for rings), but the definition works in `ordered_ring` so lemmas are proved there when that works.)

Reply:
Also, if adding the `linear_ordered_field` statement in future, presumably it should use `set.interval` from `data.set.intervals.unordered_interval`.